### PR TITLE
Improve Typescript implementation

### PR DESCRIPTION
**Pull request by realityking**
Very cool to see #147 merged. I'd like to offer a few improvements

- This removes the added `src` folder under `dist`, keeping a flatter file structure by compiling only the tests
- Includes the generated validators into type checking by providing a `.d.ts` file
- Uses `ts-node` to run tests directly of the Typescript code
- Skip generating source maps when building (they weren't shipped before but this makes it a bit more obvious what's going on)

It might be easier to review this commit by commit.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -1,17 +1,19 @@
 {
```
Comment (nam-hle) [inline tsconfig.json:1]: Please help me exclude node_modules 🥰

**Comment by nam-hle** [comment]
Great work, @realityking. Thanks a lot.

File: tsconfig.json (hunk 1 above)
Comment (realityking) [reply tsconfig.json:1]: What do you wanna achieve? Just setting `skipLibsCheck` to `true`?

Comment (nam-hle) [reply tsconfig.json:1]: I wanna speed up the compilation, but maybe lead to type riskies. So keep as it 😃

**Pull request merged**